services/server/src/server/services/utils/etherscan-util.ts

```diff
@@ -1,488 +1,110 @@
+        throw throwV2Errors
+          ? new EtherscanRequestFailedError(message)
+          : new BadGatewayError(message);
 
-export const getContractPathFromSourcesOrThrow = (
-  contractName: string,
-  sources: Sources,
-  throwV2Errors: boolean,
-): string => {
-  const contractPath = getContractPathFromSources(contractName, sources);
+      case "etherscan_missing_contract_definition":
+      case "etherscan_vyper_version_mapping_failed":
+      case "etherscan_missing_contract_in_json":
+      case "etherscan_missing_vyper_settings":
+        throw throwV2Errors
+          ? new MalformedEtherscanResponseError(message)
+          : new BadRequestError(message);
 
-  if (contractPath === undefined) {
-    const errorMessage =
-      "The sources returned by Etherscan don't include the expected contract definition.";
-    throw throwV2Errors
-      ? new MalformedEtherscanResponseError(errorMessage)
-      : new BadRequestError(errorMessage);
+      default:
+        // Fallback for any new error codes not yet handled
+        throw throwV2Errors
+          ? new EtherscanRequestFailedError(message)
+          : new BadGatewayError(message);
+    }
   }
 
-  return contractPath;
-};
-
-export const getVyperJsonInputFromSingleFileResult = (
-  etherscanResult: EtherscanResult,
-  sources: VyperJsonInput["sources"],
-): VyperJsonInput => {
-  const generatedSettings = {
-    outputSelection: {
-      "*": ["evm.deployedBytecode.object"],
-    },
-    evmVersion:
-      etherscanResult.EVMVersion !== "Default"
-        ? (etherscanResult.EVMVersion as any)
-        : undefined,
-    search_paths: ["."],
-  };
-  return {
-    language: "Vyper",
-    sources,
-    settings: generatedSettings,
-  };
-};
-
-export interface ProcessedEtherscanResult {
-  compilerVersion: string;
-  jsonInput: VyperJsonInput | SolidityJsonInput;
-  contractPath: string;
-  contractName: string;
+  // Unknown error from lib
+  throw err;
 }
 
 export const fetchFromEtherscan = async (
   sourcifyChain: SourcifyChain,
   address: string,
   userApiKey?: string,
-  throwV2Errors: boolean = false,
-): Promise<EtherscanResult> => {
-  if (!sourcifyChain.etherscanApi?.supported) {
-    const errorMessage = `Requested chain ${sourcifyChain.chainId} is not supported for importing from Etherscan.`;
-    throw throwV2Errors
-      ? new ChainNotFoundError(errorMessage)
-      : new BadRequestError(errorMessage);
-  }
-
-  let url = `https://api.etherscan.io/v2/api?chainid=${sourcifyChain.chainId}&module=contract&action=getsourcecode&address=${address}&apikey=`;
-  const apiKey =
-    userApiKey ||
-    process.env[sourcifyChain.etherscanApi.apiKeyEnvName || ""] ||
-    process.env.ETHERSCAN_API_KEY ||
-    "";
-  const secretUrl = url + apiKey;
-
-  url = url + apiKey.slice(0, 6) + "...";
-  let response;
-  logger.info("Fetching from Etherscan", {
-    url,
-    chainId: sourcifyChain.chainId,
-    address,
-  });
+  throwV2Errors = false,
```

Comment:
What was the use of this parameter? I don't see it being used anywhere

Reply:
For example in `services/server/src/server/apiv2/verification/verification.handlers.ts:132` you can see it being used. It's needed because we need to send different error responses depending on v1 vs v2